src/manager.S

```diff
@@ -129,3 +129,67 @@ add_compartment:
     b.gt      abort
 
     ret
+
+/**
+ * Function to delete an existing compartment data
+ *
+ * @param x0 Function of compartment to be deleted
+ */
+/* TODO there is a potential issue of a compartment saving a capability it was
+ * provided to it for use (i.e., PCC, DDC), such as by storing it in the memory
+ * of another compartment. If the original compartment gets deleted, and a new
+ * compartment is allocated in the same memory region, we have leaked a
+ * capability referencing the new compartment. */
+.type del_compartment, "function"
+del_compartment:
+    // Find capabilities (DDC/PCC) of given compartment
+    ldr       x1, comps_addr
+    mov       w2, #0
+    ldr       w3, comps_cnt
+
+pcc_check_start:
+    ldr       c4, [x1, #COMP_OFFSET_PCC] // get current stored PCC to check
+    gcvalue   x4, c4
+    cmp       x4, x0 // check value of current PCC against given function
+    b.eq      pcc_found
+    add       w2, w2, #1
+    cmp       w2, w3 // compartment bounds check
+    b.eq      pcc_not_found
+    add       x1, x1, #COMP_SIZE // go check next PCC
+    b         pcc_check_start
+
+pcc_not_found:
+    b         abort
+
+pcc_found:
+    // We found the address to the start of the compartment info area, at
+    // address `$x1`
+    stp       x1, lr, [sp, #-16]! // store `x1` and `lr`
+    ldr       c2, [x1, #COMP_OFFSET_DDC]
+
+    // Unmap memory region allocated
+    gcvalue   x0, c2
+    gclen     x1, c2
+    bl        munmap
+    cmp       x0, #-1
+    b.eq      abort // munmap failed
+    ldp       x0, lr, [sp], #16
```

Comment:
Given Laurie's comments about not reusing memory regions, should this remap it as `PROT_NONE` so that subsequent allocations can't accidentally reuse it?

Reply:
I think that we decided to leave the issue of reusing memory regions for a potentially more delicate solution. I personally wouldn't like to add in this model a potential DoS attack, that can be avoided by overcoming the problem "correctly".

Reply:
I'm not convinced about the DoS argument there since we don't restrict direct access to `mmap`, but we can definitely leave this question until later.

Reply:
@0152la What's the possible DoS attack angle here? I can't immediately see one, but that might just be myopia on my part!

Reply:
I think I mentioned it above, but if we allocate a region of memory, then upon deallocation, ensure that region of memory never gets reallocated again, wouldn't we eventually run out of memory?

Reply:
Ah, I'm with you! Clearly memory -- even virtual memory -- is finite in size so we can't use "fresh" memory forever. *But* virtual memory is quite big :) On most current chips it's 48 bits. That means that we will go for a long period before we exhaust virtual memory and have to reuse old memory. We could even decide at that point to do a revocation sweep (since it will happen very very rarely) to guarantee that no-one can hold a capability to a chunk of virtual memory we might want to reuse. To summarise, I don't think this is a DOS attack.

Reply:
Sure; theoretically we will run out of memory practically we won't. For this PR, do you want me to implement the non-reuse then, or amend the TODO?

Reply:
I think in this case `munmap` (rather than `mmap(PROT_NONE)`) is probably the right thing to do. Adding a todo (about revocation etc) seems sensible though.

Reply:
This [1] one here, or bring it closer to the `munmap` call here?

[1] https://github.com/capablevms/cheri-morello-compartmentalisation/pull/2/files#diff-f88500bcad8ecef5303aba1af890ac45522ccaf3d641623b6582eeacf386b0a5R138-R142